### PR TITLE
UIORGS-47 Create/Edit Contact Form

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Create/Edit Contact Form
https://issues.folio.org/browse/UIORGS-47

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/ContactPeople/ContactPersonForm/AddContactDetailsComponent.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+import {
+  Button,
+  Col,
+} from '@folio/stripes-components';
+
+import EmbeddedContactForm from './EmbeddedContactForm';
+import css from './AddContactDetailsComponent.css';
+
+class AddContactDetailsComponent extends Component {
```
Comment (aliaksei-chumakou) [inline src/ContactPeople/ContactPersonForm/AddContactDetailsComponent.js]: I don't think we use pattern to add suffix `Component` for every component

**Comment by aliaksei-chumakou** [comment]
@AndreiShumski2 I'm not sure why you've chosen to put code under `ContactPeople` module, since they are not related. Please check `contacts` module, I think it's a good place to put edit code also, since routes defined there

File: src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    formValues[name].forEach((a, i) => {
+      if (i === id) {
+        dispatch(change(`${name}[${i}].isPrimary`, true));
+      } else {
+        dispatch(change(`${name}[${i}].isPrimary`, false));
+      }
+    });
+  }
+
```
Comment (AndreiKandratsenka) [inline src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js]: What do you think about 
```suggestion
  setPrimary = (id) => () => {
    const { dispatch, change, store, name } = this.props;
    const formValues = getFormValues('ContactPersonForm')(store.getState());
    const addresses = formValues[name]
      .map((item, index) => (
        index === id
          ? { ...item, isPrimary: true }
          : { ...item, isPrimary: false }));

    dispatch(change(`${name}`, addresses));
  };

  primaryRadio = ({ input, ...props }) => {
    return (
      <label className={css.primaryToggle} htmlFor={props.fieldKey}>
        <input
          type="radio"
          checked={input.value}
          id={props.fieldKey}
          onChange={this.setPrimary(this.props.fieldKey)}
        />
        {this.props.labelForPrimaryFieldsGroup}
      </label>
    );
  };

```

File: src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          className={css.formHeader}
+          start="xs"
+        >
+          <div className={css.label}>
+            {labelForFieldsGroup}
+            {(fieldKey + 1)}
+          </div>
+          <div className={css.primaryLabel}>
+            <Field
+              component={primaryRadio}
```
Comment (AndreiKandratsenka) [inline src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js]:
```suggestion
              component={this.primaryRadio}
```

Comment (AndreiKandratsenka) [inline src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js]: and here ```
component={this.primaryRadio}
```

File: src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+              <em>
+                {this.props.emptyListMessage}
+              </em>
+            </div>
+          }
+          <ul className={css.formList}>
+            {fields.map((fieldName, i) => (
+              <li
+                className={css.formListItem}
+                key={i}
```
Comment (NikitaSedyx) [inline src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.js]: index as a key, please avoid

File: src/contacts/EditContact/EditContact.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    dispatch: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const { stripes: { store }, change, dispatch } = this.props;
+
+    return (
+      <AccordionSet>
+        <Accordion
```
Comment (NikitaSedyx) [inline src/contacts/EditContact/EditContact.js]: do we really need accordion? it was on mock up, but I think Dennis just pasted it from somewhere

File: src/ContactPeople/ContactPersonForm/AddContactDetailsComponent.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/ContactPeople/ContactPersonForm/AddContactDetailsComponent.js]: Updated

File: src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js]: Ok

File: src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.js (hunk 4 above)
Comment (AndreiShumski2) [reply src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.js]: Done. Thanks

File: src/contacts/EditContact/EditContact.js (hunk 5 above)
Comment (AndreiShumski2) [reply src/contacts/EditContact/EditContact.js]: I don't know. Let's leave it as on mock-up now

File: src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.formListItem {
+  width: 100%;
+  border-color: var(--color-border-p2);
+  border-width: 1px 1px 0;
+  border-style: solid;
+}
+
+.formListItem:last-of-type {
```
Comment (NikitaSedyx) [inline src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css]: inner selector?

File: src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+.primaryToggle {
+  padding: 4px 8px;
+  color: #555;
+  font-size: var(--font-size-medium);
+  cursor: pointer;
+  margin-bottom: 0;
+}
+
+.primaryToggle:input[type="radio"] {
```
Comment (NikitaSedyx) [inline src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css]: same

File: src/contacts/EditContact/EditContact.js (hunk 5 above)
Comment (NikitaSedyx) [reply src/contacts/EditContact/EditContact.js]: FYI Dennis closed my ticket without accordion

File: test/bigtest/tests/createContact.e2e.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  describe('Create contact form', () => {
+    it('create contact form should be present', () => {
+      expect(createContact.saveButton.isPresent).to.be.true;
+      expect(createContact.addressForm.addresses().length).to.be.equal(0);
+      expect(createContact.phoneForm.phones().length).to.be.equal(0);
+      expect(createContact.emailForm.emails().length).to.be.equal(0);
+      expect(createContact.urlForm.urls().length).to.be.equal(0);
+    });
+
+    describe('Add address', () => {
```
Comment (aliaksei-chumakou) [inline test/bigtest/tests/createContact.e2e.js]: I'm not sure it should be nested in `describe('Create contact form'`.

File: src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css (hunk 6 above)
Comment (AndreiShumski2) [reply src/contacts/EditContact/ContactDetailsForm/ContactDetailsForm.css]: Fixed

File: test/bigtest/tests/createContact.e2e.js (hunk 8 above)
Comment (AndreiShumski2) [reply test/bigtest/tests/createContact.e2e.js]: Fixed

**Pull request merged**